### ISSUE TITLE
Downstream changelog generation for empty changelog messages

### ISSUE THREAD

**Issue by wyardley** [hashicorp/terraform-provider-google#20134]
### Question

Not super important, but just for my edification: is there a reason magic modules generates a downstream changelog file for items where the message type is none / message is empty (and, in some cases, the change itself doesn't generate downstream changes other than the entry itself)? Does the tool that generate the changelog need / expect to find an entry for even PRs that don't generate an entry?

Example: https://github.com/hashicorp/terraform-provider-google/commit/ccea9d57b2d206344cbf1b0f12a4d5348c97bf46

**Comment by wyardley** [comment]
@SarahFrench @rileykarson - either of you have background or a take on this?

**Comment by rileykarson** [comment]
I don't remember personally. It may have just been easier, or generated as blank as a place to edit the changelog entry easily from the web ui.

**Comment by wyardley** [comment]
@rileykarson I guess that was my question... usually I've seen the changelog edited directly vs. in the auto-notes when it's edited after the fact - are these ever edited (in this repo) once they're generated? Usually when I've seen changelog fixes added after the fact, the PR is just updating the changelog and not the note it was generated from, but wasn't sure if there's a situation where the entire changelog gets regenerated from these files?

_If_ they're not used once they're generated the first time (e.g., for regeneration) I think the upside would be not making a commit when an (empty) changelog item is the _only_ change introduced in the provider.

**Comment by rileykarson** [comment]
No, we don't revisit them to regenerate- just during initial generation- and definitely don't use that functionality, just editing changelogs directly in the GH PR.

> I think the upside would be not making a downstream commit when an (empty) changelog item is the only change introduced in the provider.

Yeah, that'd be nice!

**Comment by wyardley** [comment]
@rileykarson is something like this along the right lines (to the point where I should open a PR or draft PR), or is this something you all would prefer to do yourselves, if at all?

```diff
--- a/.ci/magician/cmd/generate_downstream.go
+++ b/.ci/magician/cmd/generate_downstream.go
@@ -15,7 +15,7 @@ import (
 	"github.com/spf13/cobra"
 )
 
-var changelogExp = regexp.MustCompile("(?s)```release-note.*?```")
+var changelogExp = regexp.MustCompile("(?s)```release-note:(?P<noteType>[a-zA-Z]+).*```")
 
 var gdEnvironmentVariables = [...]string{
 	"BASE_BRANCH",
@@ -351,8 +351,11 @@ func addChangelogEntry(downstreamRepo *source.Repo, pullRequest *github.PullRequ
 		return err
 	}
 	rnr.Mkdir(".changelog")
-	if err := rnr.WriteFile(filepath.Join(".changelog", fmt.Sprintf("%d.txt", pullRequest.Number)), strings.Join(changelogExp.FindAllString(pullRequest.Body, -1), "\n")); err != nil {
-		return err
+	matches := changelogExp.FindStringSubmatch(pullRequest.Body)
+	if matches != nil && matches[1] != "none"
+		if err := rnr.WriteFile(filepath.Join(".changelog", fmt.Sprintf("%d.txt", pullRequest.Number)), strings.Join(changelogExp.FindAllString(pullRequest.Body, -1), "\n")); err != nil {
+			return err
+		}
 	}
 	return rnr.PopDir()
 }
```
